references/classification/train.py

```diff
@@ -80,20 +80,21 @@ def _get_cache_path(filepath):
     cache_path = os.path.expanduser(cache_path)
     return cache_path
 
-
 def main(args):
+    if args.output_dir:
+        utils.mkdir(args.output_dir)
+
+    utils.init_distributed_mode(args)
+    print(args)
+
     if args.apex:
         if sys.version_info < (3, 0):
             raise RuntimeError("Apex currently only supports Python 3. Aborting.")
         if amp is None:
             raise RuntimeError("Failed to import apex. Please install apex from https://www.github.com/nvidia/apex "
                                "to enable mixed-precision training.")
-
-    if args.output_dir:
-        utils.mkdir(args.output_dir)
-
-    utils.init_distributed_mode(args)
-    print(args)
+        if args.distributed:
+            torch.cuda.set_device(args.gpu)
```

Comment:
This is normally not needed, because this is already handled by https://github.com/pytorch/vision/blob/8d580a1f087d1758cf181269d13ed717dba0c2bd/references/classification/utils.py#L249

Reply:
correct. I've removed this.